autoprof/fit/base.py

```diff
@@ -66,13 +66,13 @@ def __init__(self, model, initial_state = None, relative_tolerance = 1e-3, **kwa
         self.loss_history = []
         self.message = ""
 
-    def fit(self):
+    def fit(self) -> None:
```

Comment:
Note `fit` returns `self` not `None`

Reply:
its a BaseOptimizer right?

Reply:
Yes, `fit` should always return a `BaseOptimizer`